File: packages/sdkwork-utils-java/src/main/java/com/sdkwork/utils/BooleanUtils.java
package com.sdkwork.utils;

import java.util.Locale;

public final class BooleanUtils {
    private BooleanUtils() {
    }

    public static Boolean parseBool(String value) {
        if (value == null) {
            return null;
        }
        switch (value.trim().toLowerCase(Locale.ROOT)) {
            case "true":
            case "1":
            case "yes":
            case "on":
                return true;
            case "false":
            case "0":
            case "no":
            case "off":
                return false;
            default:
                return null;
        }
    }

    public static boolean isTruthy(String value) {
        if (value == null || value.trim().isEmpty()) {
            return false;
        }
        switch (value.trim().toLowerCase(Locale.ROOT)) {
            case "false":
            case "0":
            case "no":
            case "off":
                return false;
            default:
                return true;
        }
    }
}

File: packages/sdkwork-utils-kotlin/src/main/kotlin/com/sdkwork/utils/BooleanUtils.kt
package com.sdkwork.utils

import java.util.Locale

object BooleanUtils {
    fun parseBool(value: String?): Boolean? {
        if (value == null) {
            return null
        }
        return when (value.trim().lowercase(Locale.ROOT)) {
            "true", "1", "yes", "on" -> true
            "false", "0", "no", "off" -> false
            else -> null
        }
    }

    fun isTruthy(value: String?): Boolean {
        if (value == null || value.trim().isEmpty()) {
            return false
        }
        return when (value.trim().lowercase(Locale.ROOT)) {
            "false", "0", "no", "off" -> false
            else -> true
        }
    }
}
